pkg/server/admin.go

```diff
@@ -573,19 +573,53 @@ func (s *adminServer) TableStats(
 }
 
 // NonTableStats is an endpoint that returns disk usage and replication
-// statistics for the time series system.
+// statistics for non-table parts of the system
 func (s *adminServer) NonTableStats(
 	ctx context.Context, req *serverpb.NonTableStatsRequest,
 ) (*serverpb.NonTableStatsResponse, error) {
-	timeSeriesStats, err := s.tableStatsForSpan(ctx, roachpb.Span{
-		Key:    keys.TimeseriesPrefix,
-		EndKey: keys.TimeseriesPrefix.PrefixEnd(),
-	})
-	if err != nil {
-		return nil, err
+	nonTableSpans := []roachpb.Span {
+		//TODO(celia) - we should probably also include stats for the [/Min, /System) span, but
+		//    we currently get stats from .Scan() which gets RangeDescriptors from Meta2.
+		// 		This won't work for [/Min, /Meta2), since this keyspace comes before Meta2 [and
+		// 		therefore won't have metadata RangeDescriptors in Meta2].
+		// 		We should probably figure out to get those stats (which may require something
+		// 		special).
```

Comment:
I bet someone from Core can give us a trick to get stats for these ranges…

Reply:
I think it should just work? By my read the scan in `tableStatsForSpan(` doesn't just scan meta2; it should scan meta1, too, if you start your scan at KeyMin. Did you try this and run into trouble?

Reply:
Thanks for the hints!

I didn't get `LocalMax` to work as a start key, but was able to get span data for `[keys.Meta2Prefix, keys.System)` -- could you confirm that this is where I want to start scanning, as well as my understanding outlined below? Thanks!

Scans starting at `keys.KeyMin` and `keys.LocalMax` were both throwing error `pkg.storage.verifyKeys(L225): "start key in [/Min,/System) must be greater than LocalMax"`.

This happened because `keys.RangeMetaKey()` maps `keys.KeyMin`, `keys.LocalMax`, and `keys.Meta1Prefix` to the `/Min` range metadata key. _[But from reading the above, my understanding is that I don't want to scan any of this, due to issues with range-local addressing]._

The first `roachpb.Key` that maps beyond `/Min` range metadata key is `keys.Meta2Prefix`, which maps to `/Meta1` range metadata key. _[My understanding is that this is where I want to start scanning, since this is out of the range-local space]._

Reply:
Hmm, `keys.LocalMax` *should* work. Maybe there's a buglet in the code you're calling? See here for an example that does include `LocalMax`:

https://github.com/cockroachdb/cockroach/blob/2b5799db2ddaac8211cd9b4128f951c8a4d71444/pkg/server/status_test.go#L500

This range-local stuff is extremely confusing and poorly designed/documented. If you have really understood it then you deserve a medal, because every time I run into it I have to spend a fair amount of time paging things back in.

The message you posted

> pkg.storage.verifyKeys(L225): "start key in [/Min,/System) must be greater than LocalMax".

indicates that you're scanning `[/Min,/System)`. I would check why that is. You should scan `[LocalMax, something)` as your "leftmost" request.